examples/generate_notices_report_for_project_version.py

```diff
@@ -53,6 +50,12 @@ def download_report(location, file_name_base, retries=10):
 					json.dump(response.json(), f, indent=3)
 			logging.info("Successfully downloaded json file to {} for report {}".format(
 					filename, report_id))
+		elif response.status_code == 412 and response.json()['errorCode'] == '{report.main.read.unfinished.report.contents}':
```

Comment:
Nice improvement. I'm testing the change on my server and seeing a different error code, so I've put a list of error codes in the cover that and will push that adaptation shortly (once the test is completed).

Reply:
Can you add this original one back to the list as well?  I pulled down the newest version with the list and mine seems to be reverting back to the original mechanism.  Which server are you testing against?  I'm going against testing.blackduck which on on 2021.2

Reply:
The original error code you had is still there. I copied it to the list of error codes and added another error code which I got from my v2020.10.1 server which I used to test the solution.

I've tested this version now against v2020.10.1 and v2021.2.0 and it works on both, i.e. it recognizes the 412 download error.

Reply:
Ah, it is kind of there.  Originally it was this: '{report.main.read.unfinished.report.contents}' but I think you accidentally copy pasted it to this: '{report.main.read.unfinished.report}'

Reply:
ah, I already merged, sorry; I'll fix on main/master branch and let you know.

Reply:
Done. Try it now and let me know.